Pegasus/sources/Geometry.cpp

```diff
@@ -6,182 +6,257 @@
+    , type(type)
 {
-    m_centerOfMass = centerOfMass;
 }
 
-glm::dvec3 const& pegasus::geometry::Shape::GetCenterOfMass() const
+pegasus::geometry::Ray::Ray()
+    : SimpleShape(SimpleShape::Type::RAY)
 {
-    return m_centerOfMass;
 }
 
-pegasus::geometry::SimpleShape::SimpleShape(glm::dvec3 const& centerOfMass, SimpleShapeType type)
-    : Shape(centerOfMass)
-    , type(type)
+pegasus::geometry::Ray::Ray(glm::dvec3 const& centerOfMass, glm::dvec3 const& normal)
+    : SimpleShape(centerOfMass, SimpleShape::Type::RAY)
+    , direction(normal)
 {
 }
 
-pegasus::geometry::Plane::Plane(glm::dvec3 const& centerOfMass, glm::dvec3 const& normal)
-    : SimpleShape(centerOfMass, SimpleShapeType::PLANE)
-    , m_normal(normal)
+pegasus::geometry::Plane::Plane()
+    : SimpleShape(SimpleShape::Type::PLANE)
 {
 }
 
-void pegasus::geometry::Plane::SetNormal(glm::dvec3 const& normal)
+pegasus::geometry::Plane::Plane(glm::dvec3 const& centerOfMass, glm::dvec3 const& normal)
+    : SimpleShape(centerOfMass, SimpleShape::Type::PLANE)
+    , normal(normal)
 {
-    m_normal = normal;
 }
 
-glm::dvec3 const& pegasus::geometry::Plane::GetNormal() const
+pegasus::geometry::Triangle::Triangle()
+    : SimpleShape(SimpleShape::Type::TRIANGLE)
 {
-    return m_normal;
 }
 
 pegasus::geometry::Triangle::Triangle(
     glm::dvec3 const& centerOfMass, glm::dvec3 const& a, glm::dvec3 const& b, glm::dvec3 const& c
 )
-    : SimpleShape(centerOfMass, SimpleShapeType::TRIANGLE)
-    , m_aVertex(a)
-    , m_bVertex(b)
-    , m_cVertex(c)
-{
-    CalculateNormal();
-}
-
-void pegasus::geometry::Triangle::SetAxes(glm::dvec3 const& a, glm::dvec3 const& b, glm::dvec3 const& c)
+    : SimpleShape(centerOfMass, SimpleShape::Type::TRIANGLE)
+    , aVertex(a)
+    , bVertex(b)
+    , cVertex(c)
 {
-    m_aVertex = a;
-    m_bVertex = b;
-    m_cVertex = c;
     CalculateNormal();
 }
 
-void pegasus::geometry::Triangle::GetAxes(glm::dvec3& a, glm::dvec3& b, glm::dvec3& c) const
-{
-    a = m_aVertex;
-    b = m_bVertex;
-    c = m_cVertex;
-}
-
-glm::dvec3 const& pegasus::geometry::Triangle::GetNormal() const
-{
-    return m_normal;
-}
-
 void pegasus::geometry::Triangle::CalculateNormal()
 {
-    m_normal = glm::cross(m_bVertex - m_aVertex, m_cVertex - m_aVertex);
+    normal = glm::cross(bVertex - aVertex, cVertex - aVertex);
 }
 
-pegasus::geometry::Sphere::Sphere(glm::dvec3 const& centerOfMass, double r)
-    : SimpleShape(centerOfMass, SimpleShapeType::SPHERE)
-    , m_radius(r)
+pegasus::geometry::Sphere::Sphere()
+    : SimpleShape(SimpleShape::Type::SPHERE)
+    , radius()
 {
 }
 
-void pegasus::geometry::Sphere::SetRadius(double r)
+pegasus::geometry::Sphere::Sphere(glm::dvec3 const& centerOfMass, double r)
+    : SimpleShape(centerOfMass, SimpleShape::Type::SPHERE)
+    , radius(r)
 {
-    m_radius = r;
 }
 
-double pegasus::geometry::Sphere::GetRadius() const
+pegasus::geometry::Cone::Cone()
+    : SimpleShape(SimpleShape::Type::CONE)
+    , radius()
```

Comment:
`apex` a.k.a. `glm::dvec3` has a default constructor that will initialize it to 0, which is precisely what I need here.